pkg/sessionctx/variable/varsutil.go

```diff
@@ -587,3 +589,24 @@ func ParseAnalyzeSkipColumnTypes(val string) map[string]struct{} {
 	}
 	return skipTypes
 }
+
+func parseSchemaCacheSize(s *SessionVars, normalizedValue string, originalValue string) (byteSize uint64, normalizedStr string, err error) {
+	defer func() {
+		if err == nil && byteSize > 0 && byteSize < (512<<20) {
+			s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.FastGenByArgs(TiDBSchemaCacheSize, originalValue))
+			byteSize = 512 << 20
+			normalizedStr = "512MB"
+		}
+		if err == nil && byteSize > math.MaxInt64 {
+			s.StmtCtx.AppendWarning(ErrTruncatedWrongValue.FastGenByArgs(TiDBSchemaCacheSize, originalValue))
+			byteSize = math.MaxInt64
+			normalizedStr = strconv.Itoa(math.MaxInt64)
```

Comment:
what's the rule that we display a number or a human reabable string?

Reply:
display what the user set. for example if user set 1073741824 we display the number of 1073741824. if user set 10GB we display 10GB